exporter/awsemfexporter/metric_translator.go

```diff
@@ -332,11 +333,40 @@ func groupedMetricToCWMeasurementsWithFilters(groupedMetric *GroupedMetric, conf
 }
 
 // translateCWMetricToEMF converts CloudWatch Metric format to EMF.
-func translateCWMetricToEMF(cWMetric *CWMetrics) *LogEvent {
+func translateCWMetricToEMF(cWMetric *CWMetrics, config *Config) *LogEvent {
 	// convert CWMetric into map format for compatible with PLE input
 	cWMetricMap := make(map[string]interface{})
 	fieldMap := cWMetric.Fields
 
+	//restore the json objects that are stored as string in attributes
+	for _, key := range config.ParseJSONEncodedAttributeValues {
+		if fieldMap[key] == nil {
+			continue
+		}
+
+		if t := reflect.TypeOf(fieldMap[key]).Name(); t != "string" {
```

Comment:
I think you can do something like `if t, ok := t.(string); ok`

Reply:
Cool. This is better.